pages/table/dynamic-content.page.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useState } from 'react';
+
+import { useCollection } from '@cloudscape-design/collection-hooks';
+
+import Button from '~components/button';
+import CollectionPreferences, { CollectionPreferencesProps } from '~components/collection-preferences';
+import Header from '~components/header';
+import Pagination from '~components/pagination';
+import Table from '~components/table';
+import TextFilter from '~components/text-filter';
+
+import { contentDisplayPreferenceI18nStrings } from '../common/i18n-strings';
+import ScreenshotArea from '../utils/screenshot-area';
+import { generateItems, Instance } from './generate-data';
+import {
+  columnsConfig,
+  contentDisplayPreference,
+  defaultPreferences,
+  EmptyState,
+  getMatchesCountText,
+  pageSizeOptions,
+  paginationLabels,
+} from './shared-configs';
+
+const allItems = generateItems();
+
+const DynamicContent = ({ text }: { text?: string }) => {
+  const [loaded, setLoaded] = useState(false);
+  useEffect(() => {
+    setTimeout(() => setLoaded(true), 2000);
```

Comment:
nit: Would it make sense to replace the timeout activation with a button activation? That would require to decouple the state so that clicking a button won't cause the re-render of the component that renders a table. An advantage is that manual and integration tests will be better straightforward.

Reply:
to make this work you can use e.g. a custom window event so that React subscriptions and memos can be avoided